### PR TITLE
Po 7280 missing 'allow additional action' flag

### PR DESCRIPTION
**Pull request by HennaParmar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PO-7280

### Change description ###
Changes Made:

ResultEntity.java - Added 5 missing database columns:
allowPaymentTerms (Boolean)
allowAdditionalAction (Boolean) - Primary field from ticket
generatesWarrant (boolean)
requiresLja (Boolean)

ResultDto.java - Added 6 missing fields (including manualEnforcement which existed in entity but not DTO):
allowPaymentTerms
allowAdditionalAction 
generatesWarrant
requiresLja
manualEnforcement
enfNextPermittedActions

ResultControllerTest.java - Updated test to include new fields

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/opal/dto/ResultDto.java [hunk 1]
```diff
@@ -85,4 +85,22 @@ public class ResultDto {
 
     @JsonProperty("requires_employment_data")
     private Boolean requiresEmploymentData;
+
+    @JsonProperty("allow_payment_terms")
+    private Boolean allowPaymentTerms;
+
+    @JsonProperty("allow_additional_action")
```
Comment (Aneesah33) [inline src/main/java/uk/gov/hmcts/opal/dto/ResultDto.java:92]: allow_additional_action is added here, but ResultDto still uses @JsonInclude(JsonInclude.Include.NON_NULL). Since results.allow_additional_action is nullable in the db schema and this field is a Boolean, it can still be left out of the GET /results/{resultId} response when the DB value is null.

File: src/test/java/uk/gov/hmcts/opal/controllers/ResultControllerTest.java [hunk 2]
```diff
@@ -68,6 +68,11 @@ void testGetResultDto_Success_WithSimpleJsonExpectation() {
             .resultTitleCy("Welsh Title")
             .resultType("TYPE1")
             .active(true)
+            .allowAdditionalAction(true)
```
Comment (Aneesah33) [inline src/test/java/uk/gov/hmcts/opal/controllers/ResultControllerTest.java:71]: Ideally this should be covered by mapper/integration tests rather than a mocked controller test. At the moment this only proves the controller returns a mocked ResultDto, not that the real GET /results/{resultId} response includes these fields.

Comment (HennaParmar) [reply src/test/java/uk/gov/hmcts/opal/controllers/ResultControllerTest.java:71]: Thanks I've now added proper mapper and integration tests that prove the real API response includes these fields, not just mocked data.

File: src/main/java/uk/gov/hmcts/opal/dto/ResultDto.java (hunk 1 above)
Comment (HennaParmar) [reply src/main/java/uk/gov/hmcts/opal/dto/ResultDto.java:92]: Thanks, good catch. I’ve updated ResultDto so allow_additional_action is explicitly included even when the mapped Boolean value is null, by adding field-level @JsonInclude(JsonInclude.Include.ALWAYS). I’ve also added a DTO serialization test to assert the response JSON contains "allow_additional_action": null when the value is null.

**Review by Aneesah33 — APPROVED** [review]
LGTM

**Review by Nipa1410 — APPROVED** [review]
